Model/ShimModel.php

```diff
@@ -32,14 +34,13 @@ class ShimModel extends Model {
 	 * @param string|null $ds
 	 */
 	public function __construct($id = false, $table = null, $ds = null) {
-		if ($warn = Configure::read('Shim.warnAboutRelationProperty')) {
-			if ($this->getAssociated()) {
-				$message = 'Relations must be defined using $this->initialized() in ' . get_class($this);
-				if (Configure::read('debug') && $warn === 'exception') {
-					throw new ShimException($message, 500);
-				}
-				trigger_error($message, E_USER_WARNING);
-			}
+		if ($this->getAssociated()) {
+			$message = 'Relations must be defined using $this->initialized() in ' . get_class($this);
+			Shim::check(Shim::RELATIONSHIPS_PROPERTIES, $message);
+		}
+		if (!empty($this->validate)) {
+			$message = 'Default validation rules must be defined in Model::validationDefault().';
+			Shim::check(Shim::VALIDATE_PROPERTY, $message);
```

Comment:
The Problem here is that you lose the stackpointer location. it wont display anymore where this was happending in code..

Reply:
I am not sure I understand. The only difference in the stacktrace after the change that I can see is that `app/Plugin/Shim/Lib/Shim.php:37` is the top line, f.ex.:
```
1) ShimModelTest::testValidateInConstuctor
Default validation rules must be defined in Model::validationDefault().

/app/Plugin/Shim/Lib/Shim.php:37
/Plugin/Shim/Model/ShimModel.php:43
/Cake/Utility/ClassRegistry.php:169
/app/Plugin/Shim/Test/Case/Model/ShimModelTest.php:311
/Cake/TestSuite/CakeTestCase.php:84
/Cake/TestSuite/CakeTestRunner.php:67
/Cake/TestSuite/CakeTestSuiteCommand.php:98
/Cake/Console/Command/TestShell.php:283
/Cake/Console/Command/TestShell.php:326
/Cake/Console/Command/TestShell.php:265
/Cake/Console/Shell.php:462
/Cake/Console/ShellDispatcher.php:219
/Cake/Console/ShellDispatcher.php:66
```

Reply:
Here is the stacktrace from a real application:
```
1) TagTest::testFindList
Model::bindModel() has been deprecated in favor of 'contain' or association methods like Model::belongsTo().

/app/Plugin/Shim/Lib/Shim.php:44
/app/Plugin/Shim/Model/ShimModel.php:994
/app/Plugin/Shim/Model/ShimModel.php:813
/app/Plugin/Shim/Model/ShimModel.php:840
/app/Plugin/Shim/Model/ShimModel.php:887
/app/Model/Tag.php:20
/Plugin/Shim/Model/ShimModel.php:59
/app/Model/AppModel.php:110
/Cake/Utility/ClassRegistry.php:169
/app/Test/Case/Model/TagTest.php:44
/Cake/TestSuite/CakeTestCase.php:84
/Cake/TestSuite/CakeTestRunner.php:67
/Cake/TestSuite/CakeTestSuiteCommand.php:98
/Cake/Console/Command/TestShell.php:283
/Cake/Console/Command/TestShell.php:326
/Cake/Console/Command/TestShell.php:265
/Cake/Console/Shell.php:462
/Cake/Console/ShellDispatcher.php:219
/Cake/Console/ShellDispatcher.php:66
```

Notice `/app/Model/Tag.php:20`.